rust_src/remacs-sys/lib.rs

```diff
@@ -341,6 +341,122 @@ pub struct CursorPos {
     vpos: c_int,
 }
 
+#[repr(C)]
+pub struct glyph {
+    /// Position from which this glyph was drawn.  If `object' below is a
+    /// Lisp string, this is an index into that string.  If it is a
+    /// buffer, this is a position in that buffer.  In addition, some
+    /// special glyphs have special values for this:
+    ///
+    /// glyph standing for newline at end of line    0
+    /// empty space after the end of the line       -1
+    /// overlay arrow on a TTY                      -1
+    /// glyph displaying line number                -1
+    /// glyph at EOB that ends in a newline         -1
+    /// left truncation glyphs:                     -1
+    /// right truncation/continuation glyphs        next buffer position
+    /// glyph standing for newline of an empty line buffer position of newline
+    /// stretch glyph at left edge of R2L lines     buffer position of newline
+    charpos: ptrdiff_t,
+
+    /// Lisp object source of this glyph.  Currently either a buffer or a
+    /// string, if the glyph was produced from characters which came from
+    /// a buffer or a string; or nil if the glyph was inserted by
+    /// redisplay for its own purposes, such as padding, truncation, or
+    /// continuation glyphs, or the overlay-arrow glyphs on TTYs.
+    object: Lisp_Object,
+
+    /// Width in pixels.
+    pixel_width: i16,
+}
```

Comment:
what about all the other fields?

Reply:
Purposefully left out due to bitfields sadly